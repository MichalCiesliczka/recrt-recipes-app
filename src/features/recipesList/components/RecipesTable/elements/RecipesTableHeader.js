import React from 'react';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';

import type { RecipesListColumns } from '../../../recipesList.types';

type Props = {
  columns: RecipesListColumns,
};

const RecipesTableHeader = ({ columns }: Props) => (
  <TableRow>
    <TableCell>
      No.
    </TableCell>
    {
      columns.map(column => (
        <TableCell
          key={column.value}
        >
          {column.label}
        </TableCell>
      ))
    }
  </TableRow>
);

export default RecipesTableHeader;
