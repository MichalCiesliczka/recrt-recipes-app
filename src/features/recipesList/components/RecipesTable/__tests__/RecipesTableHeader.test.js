import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import RecipesTableHeader from '../elements/RecipesTableHeader';

describe('<RecipesTableHeader />', () => {
  const defaultProps = {
    columns: [
      {
        label: 'Test label',
        value: 'testValue',
        isBig: false,
      },
      {
        label: 'Test label 2',
        value: 'testValue2',
        isBig: true,
      },
    ],
  };

  it('renders without crashing', () => {
    shallow(<RecipesTableHeader {...defaultProps} />);
  });

  it('should render default status', () => {
    const tree = renderer
      .create(<RecipesTableHeader {...defaultProps} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
