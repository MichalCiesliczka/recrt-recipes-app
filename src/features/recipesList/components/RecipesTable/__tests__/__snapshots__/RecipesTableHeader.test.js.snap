// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipesTableHeader /> should render default status 1`] = `
<tr
  className="MuiTableRow-root-1"
>
  <td
    aria-sort={null}
    className="MuiTableCell-root-6"
    scope={undefined}
  >
    No.
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root-6"
    scope={undefined}
  >
    Test label
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root-6"
    scope={undefined}
  >
    Test label 2
  </td>
</tr>
`;
